### DISCUSSION TITLE
Pushing a tag within an action doesn't trigger another workflow

### DISCUSSION THREAD

**Discussion by alexcroox** [community/community#25617] [Actions]
<p>I have a node script that handles auto tagging for me:</p>
<pre><code class="lang-auto">// tag-bump-version.js
[...]
const tagAndPush = `git tag -a ${version} -m "${version}" &amp;&amp; git push origin --tags ${process.env.GITHUB_SHA}:${process.env.GITHUB_REF}`

childProcess.execSync(tagAndPush, { cwd: '.' })
</code></pre>
<p>My action runs this script:</p>
<pre><code class="lang-auto">name: Tag Branch

on:
  push:
    branches:
      - master
      - development
      - sandbox
      - staging

[...]

      - name: Tag branch
        run: |
          git config --global user.name 'Github Actions'
          git config --global user.email 'alexcroox@users.noreply.github.com'
          node ./automation/tag-bump-version.js
</code></pre>
<p>This completes without errors and I can see the new tags locally so I know they’ve been pushed.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/f/ef7a71b8a479a0b45ae928c47170243dc6d4d6bd.png" data-download-href="/uploads/short-url/yawrXymEfpqUHQf0GtUMkWcSzi5.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1065835/181092215-474ab4a8-d7ad-4cc4-b8f7-8710a7203869.png" alt="image" data-base62-sha1="yawrXymEfpqUHQf0GtUMkWcSzi5" width="690" height="49" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/f/ef7a71b8a479a0b45ae928c47170243dc6d4d6bd_2_690x49.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/f/ef7a71b8a479a0b45ae928c47170243dc6d4d6bd_2_1035x73.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/e/f/ef7a71b8a479a0b45ae928c47170243dc6d4d6bd.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/f/ef7a71b8a479a0b45ae928c47170243dc6d4d6bd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1121×80 33.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>However another action I have that listens for changes in those tags, doesn’t run:</p>
<pre><code class="lang-auto">name: Development Deploy Web

on:
  push:
    tags:
      - 0.*
</code></pre>
<p>I know this action should run because when I pushed tags previously in an external CI/CD tool it triggered it correctly.</p>
<p>So my guess is my commit/push command from the development branch is missing something</p>

**Comment by airtower-luna** [comment]
<p>I can’t see it in your workflow excerpt, but it looks like you’re using the <code>GITHUB_TOKEN</code> to authorize the push. In that case it won’t trigger new workflows by design (see <a href="https://docs.github.com/en/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow">Using the <code>GITHUB_TOKEN</code> in a workflow</a>):</p>
<blockquote>
<p>When you use the repository’s <code>GITHUB_TOKEN</code> to perform tasks on behalf of the GitHub Actions app, events triggered by the <code>GITHUB_TOKEN</code> will not create a new workflow run.</p>
</blockquote>
<p>If you want to trigger a workflow, you need to create a PAT and use that to authorize the push.</p>

**Comment by alexcroox** [comment]
<p>Thanks for additional info, I’m not using GITHUB_TOKEN no;</p>
<pre><code class="lang-auto">name: Tag Branch

on:
  push:
    branches:
      - master
      - development
      - sandbox
      - staging

concurrency:
  group: tag-branch-${{ github.ref }}
  cancel-in-progress: true

jobs:
  tag_branch:
    runs-on: ${{ matrix.os }}

    strategy:
      matrix:
        os: [ubuntu-latest]
        node: [14]

    steps:
      - name: Checkout 🛎
        uses: actions/checkout@v2
        with:
          fetch-depth: 0

      - name: Setup node env 🏗
        uses: actions/setup-node@v2.4.0
        with:
          node-version: ${{ matrix.node }}

      - name: Cache node_modules
        id: cache
        uses: actions/cache@v2
        with:
          path: ./node_modules
          key: ${{ runner.os }}-node-npm-${{ secrets.ACTIONS_CACHE_VERSION }}-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-node-npm-

      - name: Install dependencies 👨🏻‍💻
        if: steps.cache.outputs.cache-hit != 'true'
        run: CYPRESS_INSTALL_BINARY=0 npm install

      - name: Post Install on cache hit
        if: steps.cache.outputs.cache-hit == 'true'
        run: npm run postinstall

      - name: Tag branch
        run: |
          git config --global user.name 'Github Actions'
          git config --global user.email 'alexcroox@users.noreply.github.com'
          node ./automation/tag-bump-version.js
</code></pre>
<p>I assumed github actions had access to push to the same repo as I’m not authing it at all as far as I can tell (no secrets called GITHUB_TOKEN or PAT either)</p>

**Comment by airtower-luna** [comment]
<p>Looks like you are in fact using the <code>GITHUB_TOKEN</code>. :slightly_smiling_face:</p>
<p>That token is automatically created for each workflow run, and <a href="https://github.com/actions/checkout"><code>actions/checkout</code></a> uses it to fetch repository data and sets the repository up to use it for pushing if you don’t provide any other token (see its <code>token</code> option). That’s why you can push without any additional secrets. :wink:</p>

**Comment by alexcroox** [comment]
<p>Just read some more, I have this set so that’s probably why:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/b/1baa58a560a41e26f2943fcba29b5ef528c1b2e2.png" data-download-href="/uploads/short-url/3WJR7XBiOkkn7kcDgXkmbJxG9AC.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1065835/181092220-983dc161-a80c-4b83-b7fc-1c0cf0897769.png" alt="image" data-base62-sha1="3WJR7XBiOkkn7kcDgXkmbJxG9AC" width="690" height="206" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/b/1baa58a560a41e26f2943fcba29b5ef528c1b2e2_2_690x206.png, https://aws1.discourse-cdn.com/github/original/3X/1/b/1baa58a560a41e26f2943fcba29b5ef528c1b2e2.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/b/1baa58a560a41e26f2943fcba29b5ef528c1b2e2.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/b/1baa58a560a41e26f2943fcba29b5ef528c1b2e2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">913×273 24.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>What’s the best way of making sure my node script doesn’t use it, and instead uses a personal access token?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I think the easiest way is to use the <code>token</code> parameter for <code>actions/checkout</code> I mentioned above. Put the PAT in a secret, and do the checkout similar to this:</p>
<pre><code class="lang-auto">- uses: actions/checkout@v2
  with:
    token: ${{ secrets.MY_PAT }}
</code></pre>

**Reply by riderx** [reply]
I wish we had better solution, i'm setting special token with all my repos

**Reply by Time0o** [reply]
Can someone explain _why_ this is necessary? Is this documented somewhere?

**Reply by iba-cai** [reply]
https://docs.github.com/en/actions/how-tos/write-workflows/choose-when-workflows-run/trigger-a-workflow#triggering-a-workflow-from-a-workflow

**Comment by alexcroox** [comment]
<p>This is now working, thank you so much!!!</p>
